### PR TITLE
Fixes to wxSysErrorMsg

### PR DESCRIPTION
**Pull request by lanurmi**

### CODE REVIEW

File: interface/wx/log.h [hunk 1]
```diff
@@ -1232,6 +1232,20 @@ unsigned long wxSysErrorCode();
...
+    wxSysErrorCode()) is used.
+
+    This overload copies the error message to the @a szBuf, of length @a len
+    characters, provided by the caller.
+
+    @see wxSysErrorCode(), wxLogSysError()
+
+    @header{wx/log.h}
+*/
+const wxChar* wxSysErrorMsg(wxChar* szBuf, size_t len, unsigned long nErrCode = 0);
```
Comment (TcT2k) [inline interface/wx/log.h]: Maybe returning a `wxString()` would be easier for the user?
Using a name like `wxSysErrorMsgStr()`

**Comment by vadz** [comment]
I agree with @TcT2k that having `wxSysErrorMsgStr()` would be better, why add new functions working with unsafe pointers in non performance-critical code?

Otherwise I agree, of course, that making this function MT-safe is nice and it would even be worth explicitly mentioning this in the documentation.

Could you please update the patch to use `wxString`? TIA!

**Comment by lanurmi** [comment]
After some further thought, why add a new function at all, as the underlying issue can also be solved by making the one and only `wxSysErrorMsg()` return `wxString`. May break compilation of legacy code in rare cases, but on the other hand it would also automatically fix potential MT-safety issues without changes to legacy code.

**Comment by vadz** [comment]
On one hand I agree that this could be acceptable for 3.2, but OTOH why bother thinking about whether it can break some non-trivially fixable code or not when we can just add the new function and keep the existing one unchanged which is guaranteed not to break anything? I'd still do it just because it's cognitively simpler...

**Comment by lanurmi** [comment]
And then there will be two functions that appear to do exactly the same thing, yet the old one should really never be used in new code, which will not be obvious. But fair enough, I'll amend the PR to add a `wxSysErrorMsgStr()`.

**Comment by vadz** [comment]
We should mention the new function in docs and the comments, I think this should make things clear enough.

TIA!

**Comment by lanurmi** [comment]
Done. Also added a commit that changes all `wxSysErrorMsg()` calls elsewhere into `wxSysErrorMsgStr()`.

**Comment by vadz** [comment]
Thanks!

**Comment by MaartenBent** [comment]
Warning caused by this PR:
`..\..\src\common\log.cpp(1095): warning C4457: declaration of 'len' hides function parameter`

**Pull request merged**